From 7e9dbc21cab1c17a5e3a5b4d402479e40bb576d2 Mon Sep 17 00:00:00 2001
From: "R, Kannappan" <r.kannappan@intel.com>
Date: Mon, 5 Oct 2015 19:27:13 +0530
Subject: [PATCH] sink_pe: handle source capabilities in all the states

except for hard_reset and transition_to_default state, when source
capability message is sent by the port partner, it should be
processed.

Change-Id: Ief4cff1de05fe4126ce94c9f1724405f5cfddeff
Signed-off-by: R, Kannappan <r.kannappan@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-19276
---
 drivers/usb/typec/pd/sink_port_pe.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/typec/pd/sink_port_pe.c b/drivers/usb/typec/pd/sink_port_pe.c
index 99e827d..f118adb 100644
--- a/drivers/usb/typec/pd/sink_port_pe.c
+++ b/drivers/usb/typec/pd/sink_port_pe.c
@@ -1019,9 +1019,11 @@ static int sink_port_policy_rcv_pkt(struct policy *p, struct pd_packet *pkt,
 
 	switch (evt) {
 	case PE_EVT_RCVD_SRC_CAP:
-		complete(&sink->wct_complete);
+		if (!completion_done(&sink->wct_complete))
+			complete(&sink->wct_complete);
 		/* Process ScrcCap if sink pe is waiting for caps */
-		if (sink->cur_state == PE_SNK_WAIT_FOR_CAPABILITIES)
+		if (sink->cur_state != PE_SNK_HARD_RESET ||
+			sink->cur_state != PE_SNK_TRANSITION_TO_DEFAULT)
 			sink_handle_src_cap(sink, pkt);
 		break;
 	case PE_EVT_RCVD_GET_SINK_CAP:
-- 
1.9.1

